src/components/settings/Settings.h

```diff
@@ -50,6 +50,17 @@ namespace Pinetime {
         int colorIndex = 0;
       };
 
+      enum class HeartRateBackgroundMeasurementInterval : uint8_t {
+        Off,
+        Continuous,
+        TenSeconds,
```

Comment:
Ten seconds is not a good option to expose: the PPG takes at least 6.4s to get a value and there's honestly no point turning the sensor off for 3.6s, just leave it on and enjoy much more stable heart rate values as the spectral averaging won't get reset every cycle

Reply:
If that's the case then it would make sense to have a different interval instead. Which interval would you propose instead?
Would 15s make more sense?

Reply:
I think having 30 as the minimum is OK. The sensor is already allowed to run for up to 30s in the background if it can't find a good signal, in which case 30s mode becomes the same as continuous

Reply:
The formatting of the settings screen only really works with 2*n items 
So it would be nice to have another value instead. 
Maybe 15s ? that would be roughly 50% off time and 50% on time for the sensor.
And longer values than 30m doesn't really make sense anymore I think, but I'm also down to add one hour as an option

Reply:
Hmm how does it look with an odd number of items? If there's a rendering bug it should be fixed, I could take a look

I misinterpreted the timings actually. It's not the measurement-measurement interval as I thought but actually the measurement finished (got value or given up) to measurement start waiting interval

Is that intended? If I enabled this feature at say 1 minute, I would expect a reading to arrive every 1 minute rather than every 1 minute 20 seconds (if it took 20s to find the HR)

If it's not intended then I think 15s is also a bit pointless, but if it is then I think it could just about make sense

There are two reasons I think short measurement times are bad:
- It takes a minimum of 6 seconds for the heart rate to be found, and often more if the signal isn't super clean
- The heart rate algorithm becomes more accurate the longer it is running

So sensor duty cycles (% of time passing with sensor on) between say 50-100% are a bit pointless IMO, just run the sensor all the time and receive the benefits of always having a value, and that value being more stable and reliable

Reply:
It is not a perfect interval setting (meaning setting it to 1 minute = one value every minute). It's just a quick and easy solution to get the feature of repetitive measurements in the background.

I think it is difficult to make it more accurate. Since it can't be determined beforehand how long the sensor needs to measure until enough data is captured right? I mean there is an upper limit of 30s (for which the heart rate task just skips the current measurement) and there is a lower limit of around 6s, but that's still a wide range of measuring time. I think the current implementation is easy and quite robust in the regards that you know "If there was a measurement, then after x time another measurement is started"
Otherwise you would need to store the timestamp of the last measurement, and predict how long the sensor needs to take for the next measurement and start preemptively like (predicted time: 10s, interval time: 15s -> actual wait time: 5s) but this just moves the possible wait time from [21s, 44s] to [11s, 34s]. So we could decide to have a hard coded "expected sensor time"
The best approach would be a running average of the last n "sensor times" which would probably be the best predictor, but this adds two additional variables and one additional magic number (window size | n), which increases the added size to the binary. It's not a lot of course, but I had to already move an integer from a instance variable to a local variable that is changed via pointer and do other things to keep the size under the limit.

I'm fine with having a more complicated algorithm, but it seems a lot of effort for probably not a small positive effect, since I assume most people don't need a very precise interval. Also it only really matters for smaller values. If you use a 5 minute interval, another 6-15s is not really that important.

I'm also assuming, that having the sensor running all the time wastes battery. If you just want a HR roughly every 15 seconds, then having 15s wait + 6s measurement means having the sensor 71% of the time off. Which is quite a lot better than having it run all the time. Even if it degrades to 15s wait + 30s measurement, it's still 33% of the time deactivated. Instead of always on.
Even if the output quality might be a bit less accurate. There is always the option of continuous mode if someone wants more and better data. 
And the code would have to decide based on the chosen interval setting if the sensor should be running all the time or have breaks in between, which makes it more complicated again.

For longer interval times, this would mean that the sensor is restarted for every measure, but then you again just get the bad "warm-up" data, so if the imprecise data that is retrieved from the sensor that has just been started is a problem, then the longer interval times shouldn't exist anyway (or the the whole PR needs to be a lot more complicated). Because having the sensor running all the time for measurements every 10 minutes seems like a waste of energy.

This PR is open quite for a long time already and I think small improvements can be discussed much more thoroughly and implemented way better after more people had the chance to test it.

But again I'm open for suggestions for making the code better and if anything (like an inaccurate interval time) is a deal breaker, I will of course fix it!

Reply:
Ah sorry, what I was thinking was actually something different. I agree that trying to predict measurement time would be a bad idea (too complex), and having cases where runtime degrades to continuous is not ideal (for longer periods).

What I meant was storing the **start** time of the measurement, and then starting the next measurement after `start+period`, regardless of whether the last measurement succeeded or failed. So if 1 minute measurement is set (and all measurements succeed), after 1 hour 60 measurements will have been made. Currently, if each measurement took 20s, only 45 measurements would be made in an hour. This does cause any measurement period <= MAX_INTERVAL (30s) to degrade to continuous when heart rate cannot be detected.

What do you think of this?

If you're hitting memory limits, shrinking the heap wouldn't be unreasonable BTW, there is currently very little static memory free and the heap will probably be shrunk by 1K or 512B soon anyway

Reply:
Yes, please start a measurement every x seconds. That is what users will expect to happen.

Reply:
Implemented as discussed